domains/runtime/evm/src/lib.rs

```diff
@@ -693,7 +699,8 @@ pub fn extract_signer(
     extrinsics
         .into_iter()
         .map(|extrinsic| {
-            let maybe_signer = extract_signer_inner(&extrinsic, &lookup);
+            let maybe_signer =
+                flatten_and_encode_signer_extraction(extract_signer_inner(&extrinsic, &lookup));
```

Comment:
`flatten_and_encode_signer_extraction` doesn't seems necessary

```rust
extract_signer_inner(&extrinsic, &lookup)
    .and_then(|res| res.ok().map(Encode::encode))
```

Reply:
This function is required as this same non-trivial operation has to be repeated in multiple places otherwise which is error prone and since the output will always be `Vec<u8>` the error won't be detected at time of compilation.

Reply:
It is just one line of code and is used in 2 places, what it is essentially doing is *ignoring the error from `extract_signer_inner` and encode the account id*, since the caller needs `Vec<u8>` the code won't compile if the caller is not doing the exact same, so I can't see how it is non-trivial and error prone.

Reply:
@NingLin-P It is used in two places and the conversion is really specific. Regarding the output, since encode can take any data type for example instead of `AccountId` even if by mistake `Option` and `Result` are passed they still will be encoded to `Vec<u8>` which will still compile. 

So, for example, if I do:
```
 extract_signer_inner(&extrinsic, &lookup)
    .and_then(|res| res.encode())
```

And 

```
 extract_signer_inner(&extrinsic, &lookup)
    .and_then(|res| res.ok().map(Encode::encode))
```


It would still compile even though the resulting `Vec<u8>` would be different. 

I understand where you are coming from, but I would prefer less error prone code over more elegant code.

Reply:
There are `Vec<u8>` usages over the codebase, it is impossible and unrealistic to rely on the compiler to capture all the errors. Here, the caller of `extract_signer_inner` has the full context and the exact idea of how to handle the result and turn it to `Vec<u8>`, I can't see how it can lead to a mistake unless someone does that intentionally.

Reply:
@NingLin-P Okay. I thought this over and it make sense to me. I will make the changes.